pkg/kudoctl/cmd/diagnostics/processing_context.go

```diff
@@ -0,0 +1,57 @@
+package diagnostics
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// processingContext - shared data for the resource collectors
+// provides property accessors allowing to define a collector before the data it needs is available
+// provides update callback functions. callbacks panic if called on a wrong type of runtime.Object
+type processingContext struct {
+	podNames      []string
+	root          string
+	opName        string
+	opVersionName string
+	instanceName  string
+}
+
+func (ctx *processingContext) attachToRoot() string {
+	return ctx.root
+}
+
+func (ctx *processingContext) attachToOperator() string {
+	return fmt.Sprintf("%s/operator_%s", ctx.root, ctx.opName)
+}
+
+func (ctx *processingContext) attachToInstance() string {
+	return fmt.Sprintf("%s/instance_%s", ctx.attachToOperator(), ctx.instanceName)
+}
+
+func (ctx *processingContext) mustSetOperatorNameFromOperatorVersion(o runtime.Object) {
```

Comment:
Lol, this method name is too long, even by Java's standards.

Reply:
It's a trade-off for generic runtime.Object.I chose a fully descriptive name for a callback. It seemed to me better than adding more code.

Reply:
```suggestion
func (ctx *processingContext) setOperatorNameFromOperatorVersion(o runtime.Object) {
```

Same here: I don't understand the "must" prefix, having just the "set..." part makes it a lot clearer on invocation:

```
callback:       ctx.setOperatorNameFromOperatorVersion,
```

This tells me: Ah, the callback will set the operator name from the operator version. Nice. :)

btw, I don't mind long descriptive names